src/TransposedSource.jl

```diff
@@ -233,6 +234,9 @@ function TransposedSource(;fullpath::Union{AbstractString,IO}="",
             columntypes[c] = typ
         end
     end
+    if !weakrefstrings
+        columntypes = [T <: WeakRefString ? String : T for T in columntypes]
```

Comment:
Since `T` could be e.g. `Union{Missing, WeakRefString{UInt8}}` (and `WeakRefStringArray` now supports that) the code here should do the similar thing as a little bit above when converting to categorical.

Reply:
The same thing should be done for `Source`.
There's also #138 to consolidate column type detection code.

Reply:
addressed